scripts/build/gen_relocate_app.py

```diff
@@ -513,27 +519,31 @@ def parse_input_string(line):
         mem_region = mem_region.rstrip()
         phdr, rest = rest.split(':', 1)
 
+    flag_list, rest = rest.split(':', 1)
+
     # Split lists by semicolons, in part to support generator expressions
-    flag_list, file_list = (lst.split(';') for lst in rest.split(':', 1))
+    file_list, symbol_filter = (lst.split(';') for lst in rest.split(',', 1))
 
-    return mem_region, phdr, flag_list, file_list
+    return mem_region, phdr, [flag_list], file_list, symbol_filter
 
 
-# Create a dict with key as memory type and files as a list of values.
+# Create a dict with key as memory type and (files, symbol_filter) tuple
+# as a list of values.
 # Also, return another dict with program headers for memory regions
 def create_dict_wrt_mem():
     # need to support wild card *
     rel_dict = dict()
     phdrs = dict()
 
-    input_rel_dict = args.input_rel_dict.read()
+    input_rel_dict = args.input_rel_dict.read().splitlines()
     if input_rel_dict == '':
```

Comment:
```suggestion
    if not input_rel_dict:
```
`input_rel_dict` is now a list.

Reply:
fixed

Reply:
think the fix for this is in wrong commit

Reply:
thanks for spotting, fixed.